File: python/inflence-on-js/examples/listparam_paramiterator.py
from paramiterator import ParamIterator


class ListParam:

    def __init__(self, csv_str):
        self.csv_str = csv_str

    def __str__(self):
        return self.csv_str

    def __iter__(self):
        return ParamIterator(self.csv_str)


for param in ListParam("un,deux,trois"):
    print(param)

File: python/inflence-on-js/examples/listparam_paramiterator.js
function paramIterator(csv_str) {
  var position = 0;
  var done = false;
  return {
    next: function() {
      if (done)
        return {done: true, value: undefined};
      var commaPosition = csv_str.indexOf(",", position);
      if (commaPosition === -1) {
        done = true;
        var value = csv_str.slice(position);
      } else {
        var value = csv_str.slice(position, commaPosition);
        position = commaPosition + 1;
      }
      return {done: false, value: value};
    }
  }
}

function listParam(csvStr) {
  return {
    csvStr: csvStr,
    [Symbol.iterator]: function() {
      return paramIterator(csvStr);
    },
    toString: function() {
      return csvStr;
    }
  }
}

var params = listParam("one,two,three");
console.log("My list param is " + params + ".");
for (var param of params)
    console.log("One of its params is " + param + ".")
